test/core/image_processing/sobel_scharr.cpp

```diff
@@ -0,0 +1,56 @@
+#include <boost/gil/image_processing/numeric.hpp>
+#include <boost/gil/detail/math.hpp>
+#include <boost/core/lightweight_test.hpp>
+
+#include <algorithm>
+
+namespace gil = boost::gil;
+
+void test_dx_sobel_kernel()
+{
+    const auto kernel = gil::generate_dx_sobel(1);
+    BOOST_TEST(std::equal(
+        kernel.begin(),
+        kernel.end(),
+        gil::dx_sobel.begin()
+    ));
```

Comment:
I think single line would be fine here, as [it can be up to 100 characters long](https://github.com/boostorg/gil/blob/develop/CONTRIBUTING.md#guidelines).